src/applications/detection/main.cpp

```diff
@@ -52,6 +53,18 @@ using namespace boost::gil;
 namespace bfs = boost::filesystem;
 
 /**
+ * @brief Check if a string is an integer number.
+ * 
+ * @param[in] s The string to check.
+ * @return Return true if the string is an integer number
+ */
+bool isInteger(std::string s)
+{
+  std::regex e("^-?\\d+");
```

Comment:
this sadly crashes at runtime with gcc4.8 (ubuntu 14.04.05) as regex apparently are not fully supported in that version. I'll find a workaround...